### PR TITLE
CU-86a5wnpn9-NEON3 - NFI Skin - Enhancement for Christmas Campaign (N3)

### PR DESCRIPTION
**Pull request by hotequil**

### CODE REVIEW

File: src/renderer/src/store/reducers/AuthReducer.ts [hunk 1]
```diff
@@ -73,6 +75,22 @@ const authReducerMigrations = {
...
+              accounts: wallet.accounts.map((account: any) => {
+                account.skin = { id: account.backgroundColor || UtilsHelper.getSkinColor(), type: 'color' }
+
+                delete account.backgroundColor
+
+                return account
+              }),
+            })),
+          },
+        },
```
Comment (dustr94) [inline src/renderer/src/store/reducers/AuthReducer.ts]: It is not necessary as backgroundColor does not exist in NWD

Don't forget to create a migration to remove lastNftSkin

File: src/renderer/src/constants/skins.tsx [hunk 2]
```diff
@@ -22,6 +23,11 @@ export const ACCOUNT_COLOR_SKINS: TAccountColorSkin[] = [
...
-
-export const LOCAL_SKINS: TLocalSkin[] = [...ACCOUNT_LOCAL_SKINS]
+export const LOCAL_SKINS: TLocalSkin[] = [
+  {
+    id: 'coz-face-december-2024',
+    blockchain: 'neo3',
+    component: <img src={CozFaceDecember2024} alt="" />,
+    unlockedContractHash: '0x76a8f8a7a901b29a33013b469949f4b08db15756',
+  },
+]
```
Comment (dustr94) [inline src/renderer/src/constants/skins.tsx]: I would leave it as it was, as there may be skins for wallets too

File: src/renderer/src/helpers/UtilsHelper.ts [hunk 3]
```diff
@@ -222,4 +223,10 @@ export class UtilsHelper {
   static isHexadecimal(hexadecimal: string) {
     return /^#([0-9a-fA-F]{3}|[0-9a-fA-F]{6})$/.test(hexadecimal)
   }
+
+  static getSkinColor(index?: number) {
+    const nextIndex = index ?? UtilsHelper.getRandomNumber(7)
```
Comment (dustr94) [inline src/renderer/src/helpers/UtilsHelper.ts]: Rename it to newIndex

File: src/renderer/src/components/AccountIcon.tsx [hunk 4]
```diff
@@ -1,49 +1,77 @@
...
-  const skin = account.skin as TNftSkin
+  const imgUrl = (account.skin as TNftSkin)?.imgUrl
+
+  if (!imgUrl) return null
+
   return (
-    <div className="w-full h-full relative bg-gray-300/30">
-      <img src={skin.imgUrl} className="w-full h-full object-cover" />
+    <div className="flex w-full h-full items-center justify-center relative bg-gray-300/30">
+      <img aria-hidden={true} src={imgUrl} alt="" className="w-full h-full object-cover absolute inset-0 m-auto" />
```
Comment (dustr94) [inline src/renderer/src/components/AccountIcon.tsx:42]: Put some alt like “NFT skin” or something similar

File: src/renderer/src/hooks/useAfterLogin.ts [hunk 5]
```diff
@@ -193,50 +193,56 @@ const useRegisterDeeplinkListeners = () => {
...
-            const hasToken = await service.nftDataService.hasToken({
-              contractHash: skin.unlockedContractHash,
-              address: account.address,
+      ownAccounts.map(
+        async account =>
+          await Promise.allSettled(
+            LOCAL_SKINS.map(async skin => {
+              try {
+                const skinId = skin.id
+                const accountBlockchain = account.blockchain
```
Comment (dustr94) [inline src/renderer/src/hooks/useAfterLogin.ts]: These two const are not necessary, just use account.blockchain and skin.id

File: src/renderer/src/hooks/useAfterLogin.ts [hunk 6]
```diff
@@ -193,50 +193,56 @@ const useRegisterDeeplinkListeners = () => {
...
-            const service = bsAggregator.blockchainServicesByName[account.blockchain]
-            if (!hasNft(service)) return
-
-            const hasToken = await service.nftDataService.hasToken({
-              contractHash: skin.unlockedContractHash,
-              address: account.address,
+      ownAccounts.map(
+        async account =>
+          await Promise.allSettled(
+            LOCAL_SKINS.map(async skin => {
```
Comment (dustr94) [inline src/renderer/src/hooks/useAfterLogin.ts]: Reverse the order of these maps, put LOCAL SKINS first, as we will have fewer skins than accounts

File: src/renderer/src/hooks/useAfterLogin.ts [hunk 7]
```diff
@@ -193,50 +193,56 @@ const useRegisterDeeplinkListeners = () => {
...
-  const { unlockedSkinIdsRef } = useUnlockedSkinIdsSelector()
-  const { accounts } = useAccountsSelector()
-  const { networkByBlockchainRef } = useSelectedNetworkByBlockchainSelector()
+const useUnlockSkins = () => {
   const dispatch = useAppDispatch()
+  const { unlockedSkinIds } = useUnlockedSkinIdsSelector()
+  const { ownAccounts } = useOwnAccountsSelector()
+
+  const unlockSkins = async () => {
+    const skinIds: string[] = []
```
Comment (dustr94) [inline src/renderer/src/hooks/useAfterLogin.ts]: Use a Set intead of an array

**Review by dustr94 — CHANGES_REQUESTED** [review]

File: src/renderer/src/components/AccountIcon.tsx (hunk 4 above)
Comment (hotequil) [reply src/renderer/src/components/AccountIcon.tsx:42]: The `aria-hidden={true}` remove the `` element from the screen reader.

**Review by dustr94 — APPROVED** [review]

**Review by thiagocbalducci — APPROVED** [review]

**Pull request merged**